Sources/SwiftParser/Modifiers.swift

```diff
@@ -43,15 +43,56 @@ extension Parser {
 
   @_spi(RawSyntax)
   public mutating func parseModifierList() -> RawModifierListSyntax? {
+    enum ExpectedTokenKind: RawTokenKindSubset {
+      case privateKeyword
+      case fileprivateKeyword
+      case internalKeyword
+      case publicKeyword
+      case staticKeyword
+      case classKeyword
+      case identifier
+
+      var rawTokenKind: RawTokenKind {
+        switch self {
+        case .privateKeyword: return .privateKeyword
+        case .fileprivateKeyword: return .fileprivateKeyword
+        case .internalKeyword: return .internalKeyword
+        case .publicKeyword: return .publicKeyword
+        case .staticKeyword: return .staticKeyword
+        case .classKeyword: return .classKeyword
+        case .identifier: return .identifier
+        }
+      }
+
+      func accepts(lexeme: Lexer.Lexeme, parser: Parser) -> Bool {
```

Comment:
This should live on an extension of the `Lookahead`.

Reply:
I assume you are worried because we do lookahead in `accepts`. So was I and I removed that feature in one of my follow-up commits in https://github.com/apple/swift-syntax/pull/718. I cherry-picked that commit to this PR now.